src/inflow.c

```diff
@@ -117,38 +108,119 @@ int inflow_readExtInflow(char* tok[], int ntoks)
     {
         basePat = project_findObject(TIMEPATTERN, tok[7]);
         if ( basePat < 0 ) return error_setInpError(ERR_NAME, tok[7]);
-    } 
+    }
+	
+	return(inflow_setExtInflow(j, param, type, tseries, basePat,
+		cf, baseline, sf));
+}
 
-    // --- include LperFT3 term in conversion factor for MASS_INFLOW
-    if ( type == MASS_INFLOW ) cf /= LperFT3;
+int inflow_validate(int param, int type, int tseries, int basePat, double *cf)
+// 
+// Purpose: Validates Inflow
+// Input:  param = -1 for Flow or Index of Pollutant
+//         type = FLOW_INFLOW, CONCEN_INFLOW or MASS_INFLOW
+//         tSeries = Time Series Index
+//         basePat = Base Pattern Index
+// Output: cf = Unit Conversion
+// Return: returns Error Code
+{
+	int errcode = 0;
+	// Validate param
+	if (param >= Nobjects[POLLUT])
+	{
+		errcode = ERR_API_POLLUT_INDEX;
+	}
+	// Validate Type
+	else if (type != FLOW_INFLOW 
+	         && type != CONCEN_INFLOW 
+	         && type != MASS_INFLOW)
+	{
+		errcode = ERR_KEYWORD;
+	}
+	// Validate Timeseries Index
+	else if (tseries >= Nobjects[TSERIES])
+	{
+		errcode = ERR_API_TSERIES_INDEX;
+	}
+	// Validate Timepattern Index
+	else if (basePat >= Nobjects[TIMEPATTERN])
+	{
+		errcode = ERR_API_PATTERN_INDEX;
+	}
+	else
+	{
+		// --- assign type & cf values for a FLOW inflow
+		if ( type == FLOW_INFLOW )
+		{
+			*cf = 1.0/UCF(FLOW);
+		}
+		// --- include LperFT3 term in conversion factor for MASS_INFLOW
+		else if ( type == MASS_INFLOW ) 
+		{
+			*cf /= LperFT3;		
+		}
+	}
```

Comment:
I agree that it's not a huge amount of validation.  I still think separating is sensible.